### PR TITLE
Change type hints in Utils.php to iterable pseudo-type

### PR DESCRIPTION
**Pull request by spawnia**
This has come up in a discussion in https://github.com/webonyx/graphql-php/pull/539.

Added strong types where possible and changed the PHPDoc to use the `iterable` pseudo-type.

### CODE REVIEW

**Comment by vladar** [comment]
Looks good to me but there are conflicts in this branch. Can you please take a look?

File: src/Utils/Utils.php [hunk 1]
```diff
@@ -199,21 +199,21 @@ public static function mapKeyValue($traversable, callable $fn)
     }
 
     /**
-     * @param mixed|Traversable $traversable
+     * @param iterable<mixed> $iterable
```
Comment (simPod) [inline src/Utils/Utils.php]: I'd also add a documentation for key everywhere (`iterable<keytype, valuetype>`)

Comment (spawnia) [reply src/Utils/Utils.php]: Feel free to add a PR.

**Comment by spawnia** [comment]
@vladar fixed.

As @simPod noted,there can be more improvements - but i think we can add them later, as needed.

**Comment by vladar** [comment]
Awesome, thanks! 🚢

**Pull request merged**